node/validator-discovery/src/lib.rs

```diff
@@ -20,29 +20,57 @@ use std::collections::HashMap;
 use std::pin::Pin;
 
 use futures::{
-	channel::mpsc,
+	channel::{mpsc, oneshot},
 	task::{Poll, self},
 	stream,
 	StreamExt,
 };
 use streamunordered::{StreamUnordered, StreamYield};
 
-use polkadot_node_subsystem::{
+use pnu_subsystem::{
 	errors::RuntimeApiError,
 	messages::{AllMessages, NetworkBridgeMessage},
 	SubsystemContext,
 };
 use polkadot_primitives::v1::{Hash, ValidatorId, AuthorityDiscoveryId, SessionIndex};
 use sc_network::PeerId;
-use crate::Error;
+
+/// Errors which can occur when connecting to validators
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
```

Comment:
why not use `pnu_subsystem_util::Error` instead of duplicating each variant? only a few variants are actually reachable

alternatively, we can change return types of `pnu_subsystem_util::request_*` to `RuntimeApiError` instead of `util::Error` and reduce the scope of errors

Reply:
I expect rustc to complain if a variant isn't used, and it isn't. That said, I'll look into reducing the number of variants.

Reply:
I think add9c59e resolves this.